src/Services/Export/WB.Services.Export.Tests/ExportJobTests/ExportArchiveHandleServiceTests.cs

```diff
@@ -119,5 +121,108 @@ public async Task Should_Delete_Archives_Based_On_CountToKeep()
             Assert.That(File.Exists(file2), Is.True);
             Assert.That(File.Exists(file3), Is.True);
         }
+        
+        [Test]
+        public async Task Should_Delete_Archives_Based_On_Both_CountToKeep_And_DaysToKeep()
+        {
+            // Arrange
+            var tenant = new TenantInfo("http://test", "testTenant");
+            this.externalArtifactsStorage.Setup(x => x.IsEnabled()).Returns(false);
+            this.fileBasedExportedDataAccessor.Setup(x => x.GetExportDirectory(tenant)).Returns(directory);
+
+            var oldFile = Path.Combine(directory, "old.zip");
+            var recentFile1 = Path.Combine(directory, "recent1.zip");
+            var recentFile2 = Path.Combine(directory, "recent2.zip");
+            var recentFile3 = Path.Combine(directory, "recent3.zip");
+
+            File.WriteAllText(oldFile, "old content");
+            File.WriteAllText(recentFile1, "recent content 1");
+            File.WriteAllText(recentFile2, "recent content 2");
+            File.WriteAllText(recentFile3, "recent content 3");
+
+            File.SetLastWriteTimeUtc(oldFile, DateTime.UtcNow.AddDays(-10));
+            File.SetLastWriteTimeUtc(recentFile1, DateTime.UtcNow.AddDays(-3));
+            File.SetLastWriteTimeUtc(recentFile2, DateTime.UtcNow.AddDays(-2));
+            File.SetLastWriteTimeUtc(recentFile3, DateTime.UtcNow.AddDays(-1));
+
+            // Act
+            await this.exportArchiveHandleService.RunRetentionPolicy(tenant, 2, 5);
+
+            // Assert
+            Assert.That(File.Exists(oldFile), Is.False); // Deleted due to daysToKeep
+            Assert.That(File.Exists(recentFile1), Is.False); // Deleted due to countToKeep
+            Assert.That(File.Exists(recentFile2), Is.True); // Kept
+            Assert.That(File.Exists(recentFile3), Is.True); // Kept
+        }
+        
+        [Test]
+        public async Task Should_Not_Delete_Archives_When_Both_CountToKeep_And_DaysToKeep_Are_Not_Set()
+        {
+            // Arrange
+            var tenant = new TenantInfo("http://test", "testTenant");
+            this.externalArtifactsStorage.Setup(x => x.IsEnabled()).Returns(false);
+            this.fileBasedExportedDataAccessor.Setup(x => x.GetExportDirectory(tenant)).Returns(directory);
+
+            var file1 = Path.Combine(directory, "file1.zip");
+            var file2 = Path.Combine(directory, "file2.zip");
+
+            File.WriteAllText(file1, "content1");
```

Comment:
use IFileSystemAccessor in ExportArchiveHandleService class
and use mock of it in tests